src/main/configService.ts

```diff
@@ -33,17 +56,49 @@ export default class ConfigService extends EventEmitter {
          * Getter and setter config listeners. 
          */
         ipcMain.on('config', (event, args) => {
-            const [fn, key] = args;
-
-            if (fn === 'get') {
-                const value = this.get(key);
-                console.log('[Config Service] Get from config store:', key, value);
-                event.returnValue = value;
-            } else
-            if (fn === 'set') {
-                const value = args[2];
-                this.set(key, value);
-                console.log('[Config Service] Set in config store:', key, value);
+            switch (args[0]) {
+                case 'get': {
+                    const value = this.get(args[1]);
+                    event.returnValue = value;
+                    return;
+                }
+
+                case 'set': {
+                    const [_, key, value] = args;
+                    if (this.configValueChanged(key, value)) {
+                        return;
+                    }
+
+                    this.set(key, value);
+                    this.emit('change', key, value);
+                    this.logConfigChanged({[key]: value});
+                    return;
+                }
+
+                case 'set_values': {
```

Comment:
Why do we need this case now?

Reply:
I think it's for multi-valued attrs, I don't think we have any of those? Maybe you've done this for future things?

Reply:
No, this is for setting an object of values directly; it's used in `src/settings/Settings.tsx` line 72, which calls `setConfigValues()` from `useSettings.ts`, which calls the IPC channel `config` with `set_values` to avoid setting them one by one and being unable to print the differences _after_ setting everything.